### DISCUSSION TITLE
No implicit 'id' in SELECT query after 1-beta.3

### DISCUSSION THREAD

**Discussion by divan** [geldata/gel#2815] [Q&A] [answered]
Hi,

so I've updated one project to use 1-beta.3 and noticed that SELECT queries stopped returning implicit 'id' fields. 

I.e. before upgrade, the code `SELECT Entity { name }` would return objects with IDs. After upgrade, those IDs are empty.

Of course, I can update all my queries, but I also can't find any documentation or mention of this change.
Was it something changed on purpose? What behavior should I expect in the future from EdgeDB ?

**Comment by fmoor** [comment, marked as answer]
> Was it something changed on purpose?

Yes, this was an intentional change. beta 3 added a feature in the server that allows language bindings to choose weather IDs are added to query results implicitly. Since go requires users to explicitly define each field on a struct it seemed good to not force users to define ID fields where that field is never used. Other more dynamic languages retained the implicit id behavior. 

> What behavior should I expect in the future from EdgeDB ?

When using the go binding all fields including `id` need to be listed explicitly in queries. The nodejs and python bindings will include the `id` field implicitly.

**Reply by divan** [reply]
Got it. Thank you!
